### PR TITLE
Fix pre-6.0 response to unknown replication actions

### PR DESCRIPTION
**Pull request by jasontedor**
When sending replica requests for replication operations, we skip sending the request to pre-6.0 nodes for operations that such nodes would not be aware of (e.g., the background global checkpoint sync, or the primary/replica resync) since they would not know what to do with these requests. Yet, we simulate that we received responses from these nodes. Today, this is done by simulating that they sent us that their local checkpoint is unassigned sequence number. However, for pre-6.0 nodes we have introduced a special local checkpoint used in the global checkpoint tracker for such nodes and that is what we should use here too. This commit fixes this issue.

Relates #10708

### CODE REVIEW

**Review by s1monw — APPROVED** [review]
LGTM

File: core/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java [hunk 1]
```diff
@@ -1023,6 +1023,12 @@ public ReplicationGroup getReplicationGroup() {
         }
 
         public ReplicaResponse(long localCheckpoint) {
+            /*
+             * A replica should always know its own local checkpoint so this should always be a valid sequence number or the pre-6.0 local
+             * checkpoint value when simulating responses to replication actions that pre-6.0 nodes are not aware of (e.g., the global
+             * checkpoint background sync, and the primary/replica resync).
+             */
+            assert localCheckpoint != SequenceNumbersService.UNASSIGNED_SEQ_NO;
```
Comment (ywelsch) [inline core/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java]: can you also add this assertion to `GlobalCheckpointTracker.updateLocalCheckpoint` (the private method)? I guess that some tests will start to fail as they are not correctly exercising the tracker.

Comment (jasontedor) [inline core/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java]: @ywelsch I pushed 78f1d4c1f9ab558d487538c3f25600f16c39469b, can you take another look?

**Review by ywelsch — APPROVED** [review]
LGTM. Thanks @jasontedor

**Comment by jasontedor** [comment]
Thanks @ywelsch and @s1monw.

**Pull request merged**